### PR TITLE
Updates and all associated links in the glossary in sdk docs v5.1.4 - Closes #1189

### LINKED ISSUE #1189: Update Glossary in SDK docs
Since the new guides/tutorials have been completed, the Glossary needs to be updated to reflect all new terms, etc.

### PR DESCRIPTION
**Pull request by chris529**
Major overhaul in the SDK glossary removed ambiguous terminologies and added in all associated links throughout the SDK docs pertaining to all relevant glossary entries.

Closes #1189

### CODE REVIEW

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 1]
```diff
@@ -201,7 +200,7 @@ npm install @liskhq/<package-name> # <1>
 
 Upon successful completion, NPM will add the Lisk Elements package to your `package.json` file.
 
-==== Load via CDN
+==== Load via CDN (Content Delivery Network)
```
Comment (Tschakki) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:203]: Would remove it either from the title , or in the description below, 1 time is enough

File: modules/ROOT/pages/advanced-explanations/communication.adoc [hunk 2]
```diff
@@ -181,7 +181,7 @@ Prefix and suffix are always separated by a colon `:`.
 
 == Interfaces
 
-A blockchain application communicates via <<actions>> and <<events>> which can be invoked (actions) or subscribed to (events) via xref:{url_glossary_rpc}[Remote-Procedure-Calls (RPCs)].
+A blockchain application communicates via <<actions>> and <<events>> which can be invoked (actions) or subscribed to (events) via RPCs (Remote-Procedure-Calls).
```
Comment (Tschakki) [inline modules/ROOT/pages/advanced-explanations/communication.adoc]: No need to explain a terminology twice on the same page, please remove

File: modules/ROOT/pages/glossary.adoc [hunk 3]
```diff
@@ -38,7 +38,7 @@ An account consists of a unique identifier / <<account-address,address>>, an amo
 
 Addresses are unique identifiers for accounts in the Lisk blockchain.
 They are the 20-byte hash value of the public key that controls the account initially.
-In all front-end products, addresses are displayed in a human-readable representation called <<Lisk32-representation>>.
+In all front-end products, addresses are displayed in a human-readable representation called <<Lisk32 representation>>.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: this anchor link doesnt look right

File: modules/ROOT/pages/glossary.adoc [hunk 4]
```diff
@@ -199,67 +170,33 @@ More details can be found in the xref:{url_hash_onion}[enable forging] section.
...
 Furthermore, a LIP also describes the requirements, rationale and motivation for the required changes.
 
+=== Lisk SDK
+
+The Lisk SDK is an open-source software development kit which enables developers to easily build scalable Blockchain applications using JavaScript.
+
 === Lisk32 representation
 
-The Lisk32 representation of <<account-addresses>> starts with the letters "lsk" followed by a custom Base32 encoding of the 20 bytes of the address appended with a BCH checksum.
+The Lisk32 representation of an <<Account Address>> starts with the letters "lsk" followed by a custom Base32 encoding of the 20 bytes of the address appended with a BCH checksum.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: wrong link format

File: modules/ROOT/pages/glossary.adoc [hunk 5]
```diff
@@ -52,10 +52,6 @@ An active delegate can mostly be considered as a forging delegate, who is able t
 
 The Advanced Encryption Standard (AES), has a key length of 256 bits and is a block cipher adopted as an encryption standard by the U.S. government for military and government use. CBC (Cipher Blocker Chaining), is an advanced form of block cipher encryption. A detailed description of how AES functions can be found {url_aes}[here^].
 
-=== API - Application Programming Interface
-
-An application programming interface is a software intermediary that allows two applications to talk to each other.
-
 == B
 
 === BCH Codes - Bose, Chaudhuri and Hocquenghem codes
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc:61]: BIP should be removed as well.

Blockchain application description should be improved, as it is very short, also, we have now a page dedicated to explain blockchain applications, the glossary should link to it.

File: modules/ROOT/pages/glossary.adoc [hunk 6]
```diff
@@ -271,23 +208,27 @@ A xref:{url_reclaim_lsk}[LSK reclaim] transaction type on the Mainnet converts a
...
 
-=== LTS - Long Term Support
-
-LTS is an acronym for long term support, and is applied to release lines that will be supported and maintained, to ensure any critical bugs will be fixed for an extended period of time.
-
 == M
 
 === Mainnet
 
 The official main network of Lisk.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: Description could be improved, as it very short. Also, we have a page in the docs explaining the different networks, the glossary should link to it.

File: modules/ROOT/pages/glossary.adoc [hunk 7]
```diff
@@ -352,60 +281,27 @@ A peer to peer network is a group of nodes (devices), that make up a decentraliz
...
-{url_rocksdb}[RocksDB^] is a high performance embedded database.
-An embedded persistent key value store for fast storage.
+=== Reducer
 
-=== RPC - Remote Procedure Call
-
-RPC is a request–response protocol.
-An RPC is initiated by the client, which sends a request message to a known remote server to execute a specified procedure with supplied parameters.
-The remote server sends a response to the client, and the application continues its process.
+A reducer is classed as a function which takes two arguments, namely an action and the previous state of the application, and returns the new state.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: I don't think this description is correct

File: modules/ROOT/pages/glossary.adoc [hunk 8]
```diff
@@ -352,60 +281,27 @@ A peer to peer network is a group of nodes (devices), that make up a decentraliz
...
-The SHA is a 256 bit algorithm designed for cryptographic security.
-
 === Standby Delegate
 
 A standby delegate, as each delegate is not an <<active-delegate>>.
 Rank 102-∞.
 
+=== State Store
+
+The state store is a data structure that holds a temporary state while processing a block.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: Description should be improved, as it is very short. Also, we are explaining the state store in the docs, the glossary can link to it.

File: modules/ROOT/pages/glossary.adoc [hunk 9]
```diff
@@ -352,60 +281,27 @@ A peer to peer network is a group of nodes (devices), that make up a decentraliz
...
+
 == T
 
 === Testnet
 
 The official testing network of Lisk.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: Description is veery short, should be improved

File: modules/ROOT/pages/glossary.adoc [hunk 10]
```diff
@@ -415,11 +311,21 @@ A token refers to LSK or any sidechain token. A  digital token is designated to
 An interaction with the Lisk blockchain which permanently writes data to it.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc:318]: Description here can be improved as well, e.g. describing which properties a transaction object has in general

File: modules/ROOT/pages/guides/app-development/asset.adoc [hunk 11]
```diff
@@ -181,7 +184,7 @@ public schema = {
 
 == Validating the asset data
 
-The optional function `validate()` validates the data of a transaction asset, before it is passed to the `apply()` function.
+The optional function `validate()` validates the data of a xref:{url_transaction_asset}[transaction asset], before it is passed to the `apply()` function.
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/asset.adoc]: no need to link again to the glossary

File: modules/ROOT/pages/guides/app-development/asset.adoc [hunk 12]
```diff
@@ -207,7 +210,7 @@ If the validation does not throw any errors, it means the validation has been su
 
 == Defining the asset logic
 
-The most important part of the module asset is the `apply()` function.
+The most important part of the xref:{url_module_asset}[module asset] is the `apply()` function.
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/asset.adoc]: no need to link again to the glossary

File: modules/ROOT/pages/guides/app-development/setup.adoc [hunk 13]
```diff
@@ -57,7 +58,7 @@ View the complete sample code of this guide on GitHub in the {url_github_guides_
 
 === Supported Operating Systems
 
-The Lisk SDK will run on both the Ubuntu and MacOS operating systems shown below:
+The xref:{url_lisk_sdk}[Lisk SDK] will run on both the Ubuntu and MacOS operating systems shown below:
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/setup.adoc]: no need to link again to the glossary

File: modules/ROOT/pages/references/application-cli.adoc [hunk 14]
```diff
@@ -90,7 +90,7 @@ EXAMPLES
 ----
 
 === console
-Lisk interactive REPL session to run commands.
+Lisk interactive REPL (Read-eval-print loop), session to run commands.
```
Comment (Tschakki) [inline modules/ROOT/pages/references/application-cli.adoc]: No need to explain it twice on the same page

File: modules/ROOT/pages/references/lisk-commander/commands.adoc [hunk 15]
```diff
@@ -68,7 +68,7 @@ EXAMPLES
 
 === Console
 
-Lisk interactive REPL session to run commands.
+Lisk interactive REPL (Read-eval-print loop), session to run commands.
```
Comment (Tschakki) [inline modules/ROOT/pages/references/lisk-commander/commands.adoc]: No need to explain it twice on the same page

File: modules/ROOT/pages/references/lisk-elements/p2p.adoc [hunk 16]
```diff
@@ -173,7 +171,7 @@ The PeerBook also manages static lists that are configured as described below:
 
 Furthermore, the Peerbook maintains a list of banned peers and creates a timeout of 24 hours for each banned peer, after which it can be unbanned.
 
-The PeerBook exposes actions to perform xref:{url_crud_glossary}[CRUD] operations based on peerId whereby the caller is agnostic to newPeers, triedPeers and their bucketing system.
+The PeerBook exposes actions to perform CRUD (Create, Read, Update & Delete), operations based on peerId whereby the caller is agnostic to newPeers, triedPeers and their bucketing system.
```
Comment (Tschakki) [inline modules/ROOT/pages/references/lisk-elements/p2p.adoc]: No need to explain it twice on the same page

File: modules/protocol/pages/appendix.adoc [hunk 17]
```diff
@@ -62,7 +62,7 @@ image::objectID.svg[objectID]
 
 === Key pair and address creation
 To create a new address, first a 12-word passphrase is generated following the specifications of {url_github_bip_39}[BIP39].
-The passphrase is then hashed using SHA-256 into a 256-bit string. The resulting hash is used as the seed to generate the [#index-private_key-1]#*private key*# and the [#index-public_key-1]#*public key*# using Ed25519 signature scheme.
+The passphrase is then hashed using the SHA-256 (Secure Hash Algorithm 256), into a 256-bit string. The resulting hash is used as the seed to generate the [#index-private_key-1]#*private key*# and the [#index-public_key-1]#*public key*# using Ed25519 signature scheme.
```
Comment (Tschakki) [inline modules/protocol/pages/appendix.adoc]: No need to explain it twice on the same page

**Review by Tschakki — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/glossary.adoc (hunk 3 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: As discussed, it is now updated 👍

File: modules/ROOT/pages/references/lisk-elements/index.adoc (hunk 1 above)
Comment (chris529) [reply modules/ROOT/pages/references/lisk-elements/index.adoc:203]: Done 👍

File: modules/ROOT/pages/advanced-explanations/communication.adoc (hunk 2 above)
Comment (chris529) [reply modules/ROOT/pages/advanced-explanations/communication.adoc]: Done, I have removed the duplicated explanation in the "Actions" paragraph which occurs after the "Interfaces" paragraph. 👍

File: modules/ROOT/pages/glossary.adoc (hunk 5 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc:61]: We agreed in the Glossary & Ambigous terminologies STM that we would keep BIP (See excel sheet in Trello card).
I have expanded the description of 'Blockchain Application' and also now added a link to the Blockcahin App page 👍

File: modules/ROOT/pages/glossary.adoc (hunk 6 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Updated the mainnet description and added a link 👍

File: modules/ROOT/pages/glossary.adoc (hunk 7 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: I have amended it, please feel free to add/amend/update if you feel there is something else we could add here..?

File: modules/ROOT/pages/tutorials/lns/1-blockchain.adoc [hunk 18]
```diff
@@ -254,10 +255,10 @@ Now import the constants into the `register` asset, and use them to check the va
 
 Validate the following:
 
-. The TTL value needs to be above the minimum defined TTL value (60 * 60).
+. The TTL (Time-to-live), value needs to be above the minimum defined TTL value (60 * 60).
 . The `registerFor` value needs to be between 1 and 5.
 . Only second level domain names can be registered.
-. Only domains with valid TLDs can be registered.
+. Only domains with valid TLDs (Top-level domain), can be registered.
```
Comment (ishantiw) [inline modules/ROOT/pages/tutorials/lns/1-blockchain.adoc]:
```suggestion
. Only domains with valid TLDs (Top-Level Domain), can be registered.
```

Comment (ishantiw) [reply modules/ROOT/pages/tutorials/lns/1-blockchain.adoc]: or `top-level domain`

File: modules/ROOT/pages/tutorials/nft.adoc [hunk 19]
```diff
@@ -759,7 +760,7 @@ module.exports = { NFTModule };
 <2> Set the module id to `1024`.
 
 The module ID has to be unique within the network.
-The minimum value for it is `1024`, as the other IDs are reserved for future default modules of the Lisk SDK.
+The minimum value for it is `1024`, as the other IDs are reserved for future default modules of the xref:{url_lisk_sdk}[Lisk SDK].
```
Comment (ishantiw) [inline modules/ROOT/pages/tutorials/nft.adoc]:
```suggestion
The minimum value for it is `1000`, as the other IDs are reserved for future default modules of the xref:{url_lisk_sdk}[Lisk SDK].
```

**Review by ishantiw — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments

File: modules/ROOT/pages/glossary.adoc (hunk 8 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Done 👍

File: modules/ROOT/pages/glossary.adoc (hunk 9 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Have now updated the Testnet description, please feel free to add/amend if necessary 👍

File: modules/ROOT/pages/glossary.adoc (hunk 10 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc:318]: I have expanded on the description, please free free to add/amend if necessary 👍

File: modules/ROOT/pages/guides/app-development/asset.adoc (hunk 11 above)
Comment (chris529) [reply modules/ROOT/pages/guides/app-development/asset.adoc]: Good point..it has now been removed 👍

File: modules/ROOT/pages/guides/app-development/asset.adoc (hunk 12 above)
Comment (chris529) [reply modules/ROOT/pages/guides/app-development/asset.adoc]: Done 👍

File: modules/ROOT/pages/guides/app-development/setup.adoc (hunk 13 above)
Comment (chris529) [reply modules/ROOT/pages/guides/app-development/setup.adoc]: Done 👍

File: modules/ROOT/pages/references/application-cli.adoc (hunk 14 above)
Comment (chris529) [reply modules/ROOT/pages/references/application-cli.adoc]: Done 👍

File: modules/ROOT/pages/references/lisk-commander/commands.adoc (hunk 15 above)
Comment (chris529) [reply modules/ROOT/pages/references/lisk-commander/commands.adoc]: Done 👍

File: modules/ROOT/pages/references/lisk-elements/p2p.adoc (hunk 16 above)
Comment (chris529) [reply modules/ROOT/pages/references/lisk-elements/p2p.adoc]: Done 👍

File: modules/protocol/pages/appendix.adoc (hunk 17 above)
Comment (chris529) [reply modules/protocol/pages/appendix.adoc]: Done 👍

File: modules/ROOT/pages/glossary.adoc (hunk 4 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Corrected 👍

**Review by ishantiw — DISMISSED** [review]

File: modules/ROOT/pages/advanced-explanations/communication.adoc (hunk 2 above)
Comment (Tschakki) [reply modules/ROOT/pages/advanced-explanations/communication.adoc]: I think the first time you explain it on line 36, so you can remove it here as well :) Also, `WebSocket` is actually more correct here, as RPC only refers to actions I think.

```suggestion
A blockchain application communicates via <<actions>> and <<events>> which can be invoked (actions) or subscribed to (events) via WebSocket.
```

File: modules/ROOT/pages/glossary.adoc [hunk 20]
```diff
@@ -413,13 +313,23 @@ A token refers to LSK or any sidechain token. A  digital token is designated to
 === Transaction
 
 An interaction with the Lisk blockchain which permanently writes data to it.
-A transaction can have multiple purposes, e.g. a balance transfer or a delegate registration.
+A transaction can have multiple purposes, e.g. a balance transfer or delegate registration, and posseses a common set of properties. Each transaction induces a specific operation on the accounts and is defined by a unique schema and transaction logic.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]:
```suggestion
A transaction can have multiple purposes, e.g. a balance transfer or delegate registration.
Each transaction contains a module ID and an asset ID, defining the transaction type.
All properties of a transaction object are described on the xref:protocol:transactions.adoc#transaction-properties[Lisk protocol > Transactions] page. The input data for the different <<module-asset, module assets>> is stored inside the <<transaction-asset>> of a transaction.
```

File: modules/ROOT/pages/glossary.adoc (hunk 10 above)
Comment (Tschakki) [reply modules/ROOT/pages/glossary.adoc:318]: I added my suggestion below!

File: modules/ROOT/pages/glossary.adoc [hunk 21]
```diff
@@ -352,59 +285,26 @@ A peer to peer network is a group of nodes (devices), that make up a decentraliz
...
-
-{url_rocksdb}[RocksDB^] is a high performance embedded database.
-An embedded persistent key value store for fast storage.
-
-=== RPC - Remote Procedure Call
-
-RPC is a request–response protocol.
-An RPC is initiated by the client, which sends a request message to a known remote server to execute a specified procedure with supplied parameters.
-The remote server sends a response to the client, and the application continues its process.
+A reducer is classed as a function that determines changes to the state of an application. It can use the previous state and action to execute another state.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]:
```suggestion
Reducers are functions which can be invoked via Remote-Procedure-Calls (RPC) by other modules.
Reducers are able to perform state changes on the blockchain.
Modules and module assets can invoke reducers through the `reducerHandler`.
See xref:introduction.adoc#reducers[Modules > Reducers] for more information.
```

File: modules/ROOT/pages/glossary.adoc [hunk 22]
```diff
@@ -352,59 +285,26 @@ A peer to peer network is a group of nodes (devices), that make up a decentraliz
...
 === Standby Delegate
 
 A standby delegate, as each delegate is not an <<active-delegate>>.
 Rank 102-∞.
 
-== T
+=== State Store
 
-=== Testnet
+The xref:{url_state_store}[state store] is a data structure that holds a temporary state while processing a block. It is used to retrieve data from the blockchain, and exposes an interface to create, restore, and finalize snapshots. It is based on the following 3 states: Accounts, Chain and Consensus.
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: That the state store exposes `an interface to create, restore, and finalize snapshots` is new to me.
Also, the third state `consensus` seems new to me, and is also not reflected in the stateStore interface:
https://lisk.com/documentation/lisk-sdk/introduction/modules.html#the-state-store

Please double check if the explanation is correct.

**Review by Tschakki — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/glossary.adoc [hunk 23]
```diff
@@ -28,8 +28,12 @@ Chris Braithwaite <christopher.braithwaite@lightcurve.io>
 :url_token_unlock: protocol:transactions.adoc#unlock
 :url_data_structure: advanced-explanations/schemas.adoc#data-types
 :url_hash_onion: guides/node-management/forging.adoc#add-forging-data-to-config
+:url_blockchain_app: introduction/blockchain-applications.adoc
+:url_mainnet: introduction/blockchain-applications.adoc
```
Comment (Tschakki) [inline modules/ROOT/pages/glossary.adoc]: Lisk Mainnet is referring to the Lisk Core mainnet, not general blockchain applications

```suggestion
:url_mainnet: master@lisk-core::index.adoc#mainnet
```

**Review by Tschakki — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/glossary.adoc (hunk 23 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Ah yes...well spotted Mona! 👍

File: modules/ROOT/pages/glossary.adoc (hunk 21 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Excellent! Many thanks Mona 👍

File: modules/ROOT/pages/glossary.adoc (hunk 22 above)
Comment (chris529) [reply modules/ROOT/pages/glossary.adoc]: Thanks, I have reworded it accordingly 👍

**Review by Tschakki — APPROVED** [review]
Thank you, LGTM :)

**Pull request merged**